src/Shared/ServerInfrastructure/HttpCharacters.cs

```diff
@@ -1,224 +1,54 @@
-            alphaNumeric[c] = true;
-        }
-        return alphaNumeric;
-    }
+    // Matches Http.Sys
+    // Matches RFC 3986 except "*" / "+" / "," / ";" / "=" and "%" HEXDIG HEXDIG which are not allowed by Http.Sys
+    private static readonly IndexOfAnyValues<char> _allowedHostChars = IndexOfAnyValues.Create("!$&'()-._~" + AlphaNumeric);
 
-    private static bool[] InitializeAuthority()
-    {
-        // Authority https://tools.ietf.org/html/rfc3986#section-3.2
-        // Examples:
-        // microsoft.com
-        // hostname:8080
-        // [::]:8080
-        // [fe80::]
-        // 127.0.0.1
-        // user@host.com
-        // user:password@host.com
-        var authority = new bool[_tableSize];
-        Array.Copy(_alphaNumeric, authority, _tableSize);
-        authority[':'] = true;
-        authority['.'] = true;
-        authority['-'] = true;
-        authority['['] = true;
-        authority[']'] = true;
-        authority['@'] = true;
-        return authority;
-    }
+    // tchar https://tools.ietf.org/html/rfc7230#appendix-B
+    private static readonly IndexOfAnyValues<char> _allowedTokenChars = IndexOfAnyValues.Create("!#$%&'*+-.^_`|~" + AlphaNumeric);
+    private static readonly IndexOfAnyValues<byte> _allowedTokenBytes = IndexOfAnyValues.Create("!#$%&'*+-.^_`|~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"u8);
 
-    private static bool[] InitializeToken()
-    {
-        // tchar https://tools.ietf.org/html/rfc7230#appendix-B
-        var token = new bool[_tableSize];
-        Array.Copy(_alphaNumeric, token, _tableSize);
-        token['!'] = true;
-        token['#'] = true;
-        token['$'] = true;
-        token['%'] = true;
-        token['&'] = true;
-        token['\''] = true;
-        token['*'] = true;
-        token['+'] = true;
-        token['-'] = true;
-        token['.'] = true;
-        token['^'] = true;
-        token['_'] = true;
-        token['`'] = true;
-        token['|'] = true;
-        token['~'] = true;
-        return token;
-    }
+    // field-value https://tools.ietf.org/html/rfc7230#section-3.2
+    // HTAB, [VCHAR, SP]
+    private static readonly IndexOfAnyValues<char> _allowedFieldChars = IndexOfAnyValues.Create("\t !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~" + AlphaNumeric);
 
-    private static bool[] InitializeHost()
-    {
-        // Matches Http.Sys
-        // Matches RFC 3986 except "*" / "+" / "," / ";" / "=" and "%" HEXDIG HEXDIG which are not allowed by Http.Sys
-        var host = new bool[_tableSize];
-        Array.Copy(_alphaNumeric, host, _tableSize);
-        host['!'] = true;
-        host['$'] = true;
-        host['&'] = true;
-        host['\''] = true;
-        host['('] = true;
-        host[')'] = true;
-        host['-'] = true;
-        host['.'] = true;
-        host['_'] = true;
-        host['~'] = true;
-        return host;
-    }
+    private static readonly IndexOfAnyValues<char> _invalidFieldChars = IndexOfAnyValues.Create(
```

Comment:
Personally, I find this a little hard to take in at a glance.  Maybe it would be helpful to add a comment like "\u0000 to \u001F, except \u0009, plus \u007F"?

Reply:
```suggestion
    // Values are [0x00, 0x1F] without 0x09 (HTAB) and with 0x7F.
    private static readonly IndexOfAnyValues<char> _invalidFieldChars = IndexOfAnyValues.Create(
```
Is it clearer / easier to read with this comment?

Reply:
LGTM.  Thanks!